### PR TITLE
Resolve provided spire-server bind address (#3421)

### LINKED ISSUE #3421: SPIRE Server does not support binding to hostname
SPIRE Server help says it supports binding to a DNS name, but it fails when passing in `localhost`, is this right?
```bash
$ bin/spire-server run -h                    
Usage of run:
  -bindAddress string
        IP address or DNS name of the SPIRE server

$ bin/spire-server run -bindAddress localhost
could not parse bind_address "localhost"
```
* **Version**: 8a5b108a8d2abbfe3f83404258b7039363a09741
* **Platform**: Darwin workstation.local 21.6.0 Darwin Kernel Version 21.6.0: Sat Jun 18 17:07:25 PDT 2022; root:xnu-8020.140.41~1/RELEASE_X86_64 x86_64
* **Subsystem**: server

### PR DESCRIPTION
**Pull request by guilhermocc**
Signed-off-by: Guilherme Carvalho <guilhermbrsp@gmail.com>

**Pull Request check list**

- [x] Commit conforms to CONTRIBUTING.md?
- [x] Proper tests/regressions included?
- [ ] Documentation updated?

**Affected functionality**
`spire-server` command "**bindAddress**" and "**bindPort**" flags/conf parse. 

**Description of change**
- Use TCP resolution to parse provided spire server address and port.
- Add an extra test case for invalid bindPort value.

**Which issue this PR fixes**
fixes #3421

### CODE REVIEW

File: cmd/spire-server/cli/run/run_test.go [hunk 1]
```diff
@@ -467,6 +467,17 @@ func TestNewServerConfig(t *testing.T) {
 				require.Nil(t, c)
 			},
 		},
+		{
+			msg:         "invalid bind_port should return an error",
```
Comment (MarcosDY) [inline cmd/spire-server/cli/run/run_test.go:471]: can you add a test case related with created issue? to verify that we are now supporting a hostname

**Review by MarcosDY — COMMENTED** [review]
Thanks!

File: cmd/spire-server/cli/run/run.go [hunk 2]
```diff
@@ -367,15 +367,11 @@ func NewServerConfig(c *Config, logOptions []log.Option, allowUnknownConfig bool
...
-	ip := net.ParseIP(c.Server.BindAddress)
-	if ip == nil {
-		return nil, fmt.Errorf("could not parse bind_address %q", c.Server.BindAddress)
-	}
-	sc.BindAddress = &net.TCPAddr{
-		IP:   ip,
-		Port: c.Server.BindPort,
+	tcpAddr, err := net.ResolveTCPAddr("tcp", fmt.Sprintf("%s:%d", c.Server.BindAddress, c.Server.BindPort))
+	if err != nil {
+		return nil, fmt.Errorf(`could not resolve TCP address "%q:%d": %w`, c.Server.BindAddress, c.Server.BindPort, err)
```
[the remark this answers is not in the document]
Comment (guilhermocc) [reply cmd/spire-server/cli/run/run.go]: Forgot to change it, fixed here a5b251e

File: cmd/spire-server/cli/run/run.go [hunk 3]
```diff
@@ -367,15 +367,11 @@ func NewServerConfig(c *Config, logOptions []log.Option, allowUnknownConfig bool
...
 	}
 
-	ip := net.ParseIP(c.Server.BindAddress)
-	if ip == nil {
-		return nil, fmt.Errorf("could not parse bind_address %q", c.Server.BindAddress)
-	}
-	sc.BindAddress = &net.TCPAddr{
-		IP:   ip,
-		Port: c.Server.BindPort,
+	tcpAddr, err := net.ResolveTCPAddr("tcp", fmt.Sprintf("%s:%d", c.Server.BindAddress, c.Server.BindPort))
```
Comment (azdagron) [inline cmd/spire-server/cli/run/run.go:370]: nitpick: how do you feel about renaming `tcpAddr` to `bindAddress`?

Comment (guilhermocc) [reply cmd/spire-server/cli/run/run.go:370]: In the end, this value happens to be the server config **BindAdress**, right? I like the suggestion :+1:

**Review by azdagron — APPROVED** [review]

**Pull request merged**